src/dmd/doc.d

```diff
@@ -2485,6 +2528,239 @@ private struct MarkdownDelimiter
     char type;      /// the type of delimiter, defined by its starting character
 }
 
+/****************************************************
+ * Info about a Markdown list.
+ */
+private struct MarkdownList
+{
+    string orderedStart;    /// an optional start number--if present then the list starts at this number
+    size_t iStart;          /// the index where the list item starts
+    size_t iContentStart;   /// the index where the content starts after the list delimiter
+    int delimiterIndent;    /// the level of indent the list delimiter starts at
+    int contentIndent;      /// the level of indent the content starts at
+    int macroLevel;         /// the count of nested DDoc macros when the list is started
+    char type = 0;          /// the type of list, defined by its starting character
+
+    /// whether this describes a valid list
+    @property bool isValid() inout { return type != 0; }
+
+    /****************************************************
+     * Try to parse a list item, returning whether successful.
+     * Params:
+     *  buf           = an OutBuffer containing the DDoc
+     *  iLineStart    = the index within `buf` of the first character of the line
+     *  i             = the index within `buf` of the potential list item
+     * Returns: the parsed list item. Its `isValid` property describes whether parsing succeeded.
+     */
+    static MarkdownList parseItem(OutBuffer *buf, size_t iLineStart, size_t i)
+    {
+        MarkdownList list;
```

Comment:
I suggest you skip this variable an just return what the functions you're calling return.